tests/Feature/AdmissionTest.php

```diff
@@ -0,0 +1,316 @@
+        $angol = Workshop::firstWhere('name', Workshop::ANGOL);
+
+        $user->addRole(Role::get(Role::WORKSHOP_LEADER), $aurelion);
+        $user->addRole(Role::get(Role::WORKSHOP_LEADER), $info);
+
+        $this->createApplicant("applicant1", false, [$info->id]);
+        $this->createApplicant("applicant2", true, [$aurelion->id, $info->id]);
+        $this->createApplicant("applicant3", true, [$info->id, $angol->id]);
+        $this->createApplicant("applicant4", true, [$angol->id]);
+
+        $response = $this->get(route('admission.applicants.index') . "?show_not_submitted=true");
+
+        $response->assertDontSee('applicant1');
+        $response->assertSee('applicant2');
+        $response->assertSee('applicant3');
+        $response->assertDontSee('applicant4');
+    }
+
+    /**
+     * Test viewing all finalised applications in a workshop as workshop admin.
+     *
+     * @return void
+     */
+    public function test_view_applications_as_workshop_admin()
+    {
+        $user = User::factory()->create(['verified' => true]);
+        $this->actingAs($user);
+
+        $aurelion = Workshop::firstWhere('name', Workshop::AURELION);
+        $info = Workshop::firstWhere('name', Workshop::INFORMATIKA);
+        $angol = Workshop::firstWhere('name', Workshop::ANGOL);
+
+        $user->addRole(Role::get(Role::WORKSHOP_ADMINISTRATOR), $aurelion);
+        $user->addRole(Role::get(Role::WORKSHOP_ADMINISTRATOR), $info);
+
+        $this->createApplicant("applicant1", false, [$info->id]);
+        $this->createApplicant("applicant2", true, [$aurelion->id, $info->id]);
+        $this->createApplicant("applicant3", true, [$info->id, $angol->id]);
+        $this->createApplicant("applicant4", true, [$angol->id]);
+
+        $response = $this->get(route('admission.applicants.index') . "?show_not_submitted=true");
+
+        $response->assertDontSee('applicant1');
+        $response->assertSee('applicant2');
+        $response->assertSee('applicant3');
+        $response->assertDontSee('applicant4');
+    }
+
+    /**
+     * Test viewing all finalised applications as aggregated committe member.
+     *
+     * @return void
+     */
+    public function test_view_applications_as_aggregated_committee_member()
+    {
+        $user = User::factory()->create(['verified' => true]);
+        $this->actingAs($user);
+
+        $aurelion = Workshop::firstWhere('name', Workshop::AURELION);
+        $info = Workshop::firstWhere('name', Workshop::INFORMATIKA);
+        $angol = Workshop::firstWhere('name', Workshop::ANGOL);
+
+        $user->addRole(Role::get(Role::AGGREGATED_APPLICATION_COMMITTEE_MEMBER));
+
+        $this->createApplicant("applicant1", false, [$info->id]);
+        $this->createApplicant("applicant2", true, [$aurelion->id, $info->id]);
+        $this->createApplicant("applicant3", true, [$info->id, $angol->id]);
+        $this->createApplicant("applicant4", true, [$angol->id]);
```

Comment:
Could we put this into a separate private function? It is used quite often.

Reply:
Yes, but every time it's a little different. For tests, I think this is easier to debug. I already have a createApplicant helper :D

Reply:
OK then:)